source/module_cell/print_cif.cpp

```diff
@@ -0,0 +1,169 @@
+        = acos((a1 * a2) / (a1.norm() * a2.norm())) / ModuleBase::PI * 180.0;
+    ofs << "_cell_angle_alpha " << angle_alpha << std::endl;
+    ofs << "_cell_angle_beta " << angle_beta << std::endl;
+    ofs << "_cell_angle_gamma " << angle_gamma << std::endl;
+    ofs << std::endl;
+    //ofs << "_symmetry_space_group_name_"
+    //     << this->symm.spgname << " " << std::endl;
+    ofs << "loop_" << std::endl;
+        ofs << "_symmetry_equiv_pos_as_xyz" << std::endl;
+    // ofs << "_symmetry_Int_Tables_number"
+    //     << " " << std::endl;
+    if (this->symm.symm_flag == 1)
+    {
+        for (int k = 0; k < this->symm.nrotk; ++k)
+        {
+            // Assuming ModuleBase::Matrix3 and ModuleBase::Vector3<double> are defined elsewhere
+            ModuleBase::Matrix3 gmatrix = this->symm.gmatrix[k];
+            ModuleBase::Vector3<double> gtrans = this->symm.gtrans[k];
+
+            // Cast the matrix elements to int
+            int e11 = static_cast<int>(gmatrix.e11);
+            int e12 = static_cast<int>(gmatrix.e12);
+            int e13 = static_cast<int>(gmatrix.e13);
+            int e21 = static_cast<int>(gmatrix.e21);
+            int e22 = static_cast<int>(gmatrix.e22);
+            int e23 = static_cast<int>(gmatrix.e23);
+            int e31 = static_cast<int>(gmatrix.e31);
+            int e32 = static_cast<int>(gmatrix.e32);
+            int e33 = static_cast<int>(gmatrix.e33);
+
+            ofs << "'";
+            // First column
+            if (e11 != 0) ofs << (e11 == 1 ? "x" : (e11 == -1 ? "-x" : std::to_string(e11) + "x"));
+            if (e21 != 0) ofs << (e21 > 0 ? "+" : "") << (e21 == 1 ? "y" : (e21 == -1 ? "-y" : std::to_string(e21) + "y"));
+            if (e31 != 0) ofs << (e31 > 0 ? "+" : "") << (e31 == 1 ? "z" : (e31 == -1 ? "-z" : std::to_string(e31) + "z"));
+            if (gtrans.x != 0) ofs << (gtrans.x > 0 ? "+" : "") << floatToFraction(gtrans.x);
+            ofs << ", ";
+
+            // Second column
+            if (e12 != 0) ofs << (e12 == 1 ? "x" : (e12 == -1 ? "-x" : std::to_string(e12) + "x"));
+            if (e22 != 0) ofs << (e22 > 0 ? "+" : "") << (e22 == 1 ? "y" : (e22 == -1 ? "-y" : std::to_string(e22) + "y"));
+            if (e32 != 0) ofs << (e32 > 0 ? "+" : "") << (e32 == 1 ? "z" : (e32 == -1 ? "-z" : std::to_string(e32) + "z"));
+            if (gtrans.y != 0) ofs << (gtrans.y > 0 ? "+" : "") << floatToFraction(gtrans.y);
+            ofs << ", ";
+
+            // Third column
+            if (e13 != 0) ofs << (e13 == 1 ? "x" : (e13 == -1 ? "-x" : std::to_string(e13) + "x"));
+            if (e23 != 0) ofs << (e23 > 0 ? "+" : "") << (e23 == 1 ? "y" : (e23 == -1 ? "-y" : std::to_string(e23) + "y"));
+            if (e33 != 0) ofs << (e33 > 0 ? "+" : "") << (e33 == 1 ? "z" : (e33 == -1 ? "-z" : std::to_string(e33) + "z"));
+            if (gtrans.z != 0) ofs << (gtrans.z > 0 ? "+" : "") << floatToFraction(gtrans.z);
+            ofs << "'" << std::endl;
+        }
```

Comment:
I am so amazed you have so much perseverance to implement the symmetrized version of cif!